src/v/cloud_storage_clients/abs_client.cc

```diff
@@ -724,13 +801,154 @@ ss::future<> abs_client::do_delete_object(
 
 ss::future<result<abs_client::delete_objects_result, error_outcome>>
 abs_client::delete_objects(
+  const bucket_name& bucket,
+  std::vector<object_key> keys,
+  ss::lowres_clock::duration timeout) {
+    if (!_adls_client) {
+        return do_delete_objects(bucket, std::move(keys), timeout);
+    } else {
+        return do_delete_paths(bucket, std::move(keys), timeout);
+    }
+
+    const std::string boundary = http::random_multipart_boundary();
+}
+
+ss::future<result<abs_client::delete_objects_result, error_outcome>>
+abs_client::do_delete_objects(
+  const bucket_name& bucket,
+  std::vector<object_key> keys,
+  ss::lowres_clock::duration timeout) {
+    const auto boundary = http::random_multipart_boundary();
+    auto request_body = co_await make_delete_objects_payload(
+      bucket, boundary, keys);
+
+    auto header = _requestor.make_batch_request(
+      boundary, request_body.size_bytes());
+    if (!header) {
+        vlog(
+          abs_log.warn, "Failed to create request header: {}", header.error());
+        throw std::system_error(header.error());
+    }
+
+    auto istream = make_iobuf_input_stream(std::move(request_body));
+    auto response_stream = co_await _client.request(
+      std::move(header.value()), istream, timeout);
+
+    co_await response_stream->prefetch_headers();
+    vassert(response_stream->is_header_done(), "Header is not received");
+
+    const auto status = response_stream->get_headers().result();
+    if (status != boost::beast::http::status::accepted) {
+        iobuf buf = co_await util::drain_response_stream(response_stream);
+        const auto content_type = get_response_content_type(
+          response_stream->get_headers());
+        throw parse_rest_error_response(content_type, status, std::move(buf));
+    }
+
+    auto response_istream = response_stream->as_input_stream();
+
+    co_return co_await parse_delete_objects_payload(
+      std::move(response_istream), boundary, keys);
```

Comment:
Do you intend to use the boundary returned by ContentType as per the discussion yesterday?